core/src/main/java/org/infinispan/factories/AbstractComponentRegistry.java

```diff
@@ -278,6 +298,10 @@ public void start() {
 
    protected abstract void postStart();
 
+   protected CompletionStage<Void> await() {
```

Comment:
A more descriptive name would be better here. I thought this was similar to lock await method or something or waiting for something to finish, not just returning a value.